compose/ui/ui-test-junit4/src/desktopTest/kotlin/androidx/compose/ui/test/TestBasicsTest.kt

```diff
@@ -239,4 +240,45 @@ class TestBasicsTest {
         waitForIdle()
         assertEquals(initialTime, mainClock.currentTime)
     }
+
+    @Test
+    fun runOnIdleExecutesOnUiThread() = runComposeUiTest {
+        lateinit var mainThread: Thread
+        setContent {
+            mainThread = Thread.currentThread()
+        }
+        runOnIdle {
+            assertEquals(Thread.currentThread(), mainThread)
+        }
+    }
+
+    @Test
+    fun runOnIdleExecutesWhenIdle() = runComposeUiTest {
```

Comment:
In another PR you mentioned you are going to commonize some of the new tests. Would be nice to include these too (probably excluding the one explicitly using `Thread`)

Reply:
Will do.